pinot-common/src/main/java/org/apache/pinot/common/utils/LogicalTableConfigUtils.java

```diff
@@ -139,13 +139,22 @@ public static void validateLogicalTableConfig(
           "Invalid logical table. Reason: 'physicalTableConfigMap' should not be null or empty");
     }
 
+    String databaseName = DatabaseUtils.extractDatabaseFromFullyQualifiedTableName(logicalTableConfig.getTableName());
     Set<String> offlineTableNames = new HashSet<>();
     Set<String> realtimeTableNames = new HashSet<>();
 
     for (Map.Entry<String, PhysicalTableConfig> entry : logicalTableConfig.getPhysicalTableConfigMap().entrySet()) {
       String physicalTableName = entry.getKey();
       PhysicalTableConfig physicalTableConfig = entry.getValue();
 
+      // validate database name matches
+      String physicalTableDatabaseName = DatabaseUtils.extractDatabaseFromFullyQualifiedTableName(physicalTableName);
+      if (!StringUtils.equalsIgnoreCase(databaseName, physicalTableDatabaseName)) {
+        throw new IllegalArgumentException(
+            "Invalid logical table. Reason: 'physicalTableName' should have the same database name as logical table: "
```

Comment:
> Reason: 'physicalTableName' should have the same database name as logical table

There doesn't seem to be any config key defined as `physicalTableName`?

Reply:
I will replace it with actual table name.